org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/refactoring/rename/RippleMethodFinder2.java

```diff
@@ -98,7 +98,8 @@ public void init(IType type) {
 		public IType find(IType element) {
 			IType root= element;
 			IType rep= fElementToRepresentative.get(root);
-			while (rep != null && ! rep.equals(root)) {
+			Set<IType> visited=new HashSet<>();
```

Comment:
I'm not familiar with the code.  The change is fine for the meantime and much better than an infinite loop.  The real error is somehow the fElementToRepresentative map is being formed incorrectly.  Whether designating the first representative found as the root causes other problems or not is a future question.

Reply:
> The change is fine for the meantime and much better than an infinite loop

i like that attitude